sources/packages/backend/apps/api/src/route-controllers/student/student.external.controller.service.ts

```diff
@@ -38,93 +39,25 @@ export class StudentExternalControllerService {
-            ? getISODateOnlyString(application.applicationStatusUpdatedOn)
-            : null,
-        withdrawalDate: scholasticStanding?.submittedData?.dateOfWithdrawal,
-        withdrawalReason: scholasticStanding?.changeType,
-        immigrationStatus: workflowData.studentData.citizenship,
-        bcResidency: workflowData.studentData.bcResident,
-        maritalStatus: workflowData.studentData.relationshipStatus,
-        income: workflowData.studentData.taxReturnIncome,
-        livingArrangement: ["SDA", "SIA"].includes(
-          workflowData.dmnValues.livingCategory,
-        )
-          ? "Away"
-          : "Home",
-        estimatedTotalAward: getTotalDisbursementAmountFromSchedules(
-          assessment.disbursementSchedules,
-        ),
-        dependants: application.dependants?.map(
-          (dependant: { fullName: string; dateOfBirth: string }) => ({
-            fullName: dependant.fullName,
-            birthDate: dependant.dateOfBirth,
-          }),
-        ),
-        program: {
-          startDate: offering.studyStartDate,
-          endDate: offering.studyEndDate,
-          lengthInWeeks: offering.studyBreaks.totalFundedWeeks,
-          courseLoad: "FT",
-        },
-        institution: {
-          locationCode: institutionLocation.institutionCode,
-          locationName: institutionLocation.name,
-          primaryContactFirstName: institutionLocation.primaryContact.firstName,
-          primaryContactLastName: institutionLocation.primaryContact.lastName,
-          primaryContactEmail: institutionLocation.primaryContact.email,
-          primaryContactPhone: institutionLocation.primaryContact.phone,
-        },
-        costs: {
-          tuition: assessmentData.tuitionCost,
-          booksAndSupplies: assessmentData.booksAndSuppliesCost,
-          exceptionalExpenses: assessmentData.exceptionalEducationCost,
-          livingAllowance: assessmentData.livingAllowance,
-          secondResidence: assessmentData.secondResidenceCost,
-          childCare: assessmentData.childcareCost,
-          alimony: assessmentData.alimonyOrChildSupport,
-          totalTransportation: assessmentData.transportationCost,
-          totalNeed: assessmentData.provincialAssessmentNeed,
-        },
-        disbursements: assessment.disbursementSchedules.flatMap(
-          (disbursement) => {
-            return disbursement.disbursementValues.map((disbursementValue) => ({
-              awardCode: disbursementValue.valueCode,
-              awardAmount:
-                disbursement.disbursementScheduleStatus ===
-                DisbursementScheduleStatus.Sent
-                  ? disbursementValue.effectiveAmount
-                  : disbursementValue.valueAmount,
-              fundingDate: disbursement.disbursementDate,
-              requestDate: getISODateOnlyString(disbursement.dateSent),
-            }));
-          },
-        ),
-      });
+    if (studentApplications?.length) {
```

Comment:
I believe that we have been using map over `for` when possible.
Both arrays can not be optional since they will be empty or have some value.
Not a blocker but the map considering the above could be as below.
```ts
const applications = studentApplications.map((application) =>
      this.transformApplicationDetails(application),
    );
const legacyApplications = legacyStudentApplications.map((application) =>
  this.transformLegacyApplicationDetails(application),
);
return applications.concat(legacyApplications);
```